### ISSUE TITLE
Improve left side bar

### ISSUE THREAD

**Issue by fosterlynn** [lightningrodlabs/rea-playspace#37]
This is initially an issue for discussion to figure out the best solutions, I'm just kicking it off.

The issue we'll probably run into quickest is there will be a quantity of data that wants to be on the left side bar. I'm not sure of the cleanest way to make this the most usable, in particular in finding the one you want to drag and drop.

One thing would be to move the Agent creation to somewhere else, since we're not dragging and dropping them any more.

One thought on the specifications is to create a classification/category level above them, and have an explorer type thing with expand/contract.  Leanne did this on her initial spreadsheet.  We don't know yet how specific they will want some of the resource or process specifications, we'll try to get an idea from them.
![Screenshot from 2022-06-06 13-22-05](https://user-images.githubusercontent.com/3776081/172222655-92497e3d-4326-4699-b700-61147f905720.png)

A small question:  Is there a better display term than Specification?  Type maybe?  I'll ask the users when they get more of a look.  (At least on my screen, that would bring it to one line.)

**Comment by adaburrows** [comment]
Since I updated the styles, they all fit on one line on my screen (not that it matters too much since we'll change it up soon).

Since we're going to have a menu up top that has "Map" and "Events", we could definitely add an "Agents" screen that just lists the agents. On the other hand, Eric wanted to have this feel more like a playspace, and less like enterprise software where there's a separate configuration screen. We can talk about it tomorrow.

I think there's definitely some interplay between this and our desire to overload the concept of ResourceSpecification versus EconomicResource. We settled on calling each of these things "Resource Specifications" because that's what Valueflows calls it, but if we want to just make it more friendly, we could say "Kinds of Resources" and "Kinds of Processes" then it could be clear that a "Kind of *" maps to a "*Specification".

What I'm most curious about is if this will be implemented through the `resourceClassifiedAs` field and how we want to make a UI for defining that taxonomy. It's pretty simple to define a list of classifications (labor, material, education/knowledge/document, elements, space, token) and then allow those to be referenced as a list in the resourceClassifiedAs field.

There's other issues I can see with this, too. OM2 has currency symbols defined for certain currencies, but I'm certain it doesn't have the units and symbols defined for all of the tokens or badges. I can also see that certain units will likely be associated with certain taxa in the taxonomy and that is another layer outside of OM2. I'm sure that we can just allow defining custom units for the tokens without having to submit anything to OM2 so that's not too big of an issue, but we don't have an interface for that yet.

**Comment by fosterlynn** [comment]
>we could say "Kinds of Resources" and "Kinds of Processes" then it could be clear that a "Kind of *" maps to a "*Specification"

That seems more friendly.

>What I'm most curious about is if this will be implemented through the resourceClassifiedAs field and how we want to make a UI for defining that taxonomy. It's pretty simple to define a list of classifications (labor, material, education/knowledge/document, elements, space, token) and then allow those to be referenced as a list in the resourceClassifiedAs field.

Yes, good question.  Your suggestion is what VF has as of now to do this, and it seems like a reasonable way to do it.  It also assumes that if you want more of a hierarchy of classifications, that is defined elsewhere as a taxonomy, and you can travel up from the classification definition.  But we could directly support it too, if needed, which it isn't yet. 

We can hold off for now until the users can see it and are able to provide feedback more easily.

>There's other issues I can see with this, too. OM2 has currency symbols defined for certain currencies, but I'm certain it doesn't have the units and symbols defined for all of the tokens or badges. 

Currencies (of all kinds) caused quite a lot of discussion in VF.  Where we got to is that semantically currencies are better as resource specifications, not units.*  VF itself doesn't actually care, but it seems like everyone should try to do it the same way, for interoperability, so we felt like we should make a recommendation, as well as be consistent in our examples.  The fact that currently people can invent whatever they want for currencies does tend to validate that decision.  But it is also probably less intuitive for users.  Like you think "100 dollars" and it seems like a unit.  (It is not helped by the "one" label for the one-dimensional unit that people not in the sciences would call "each" or "count" or "pieces" or similar.  But that's really a broader separate issue.)

*This begs the question, then what is your bank account?  We would support that by the containedIn relationship, so your bank account can be a resource with an account number that contains a dollars resource, or your bitcoin wallet resource contains bitcoins.  Not tried yet in practice that I know of, with this exact model.

>I'm sure that we can just allow defining custom units for the tokens without having to submit anything to OM2 so that's not too big of an issue, but we don't have an interface for that yet.

That's another alternative.  I'd prefer to try currencies as resource specs first and see.

**Comment by adaburrows** [comment]
There's a lot ideas floating around about the possible changes in #164, and this is one of the things that is hitching a ride with those ideas.

One of the ideas I had was to consolidate everything that isn't dragged-and-dropped into a tree view under the pallet (and the drag-and-drop might event go away in favor of a context menu that lets people click and add so the mouse movement is minimized).

Basically, you'd have all the Agents, Resource Specifications, Process Specifications, Actions, and even Units available to browse, edit, and add.

We could potentially even make it a feature to be able to create the kind of hierarchy of classification you've talked about above. Then would could refine it based on operator feedback (🎶 *beep boop* 🎵).
